offscreen_render_target/offscreen_render_target.cpp

```diff
@@ -173,23 +174,32 @@ namespace bnb::oep
 
         using ns = bnb::oep::interfaces::rotation;
         int32_t draw_indent{0};
+        bool before_update_swap_sizes = m_swap_sizes;
         switch (orient) {
             case ns::deg0:
                 draw_indent = 0 * drawing_plane_vert_count;
+                m_swap_sizes = false;
                 break;
             case ns::deg90:
                 draw_indent = 1 * drawing_plane_vert_count;
+                m_swap_sizes = true;
                 break;
             case ns::deg180:
                 draw_indent = 2 * drawing_plane_vert_count;
+                m_swap_sizes = false;
                 break;
             case ns::deg270:
                 draw_indent = 3 * drawing_plane_vert_count;
+                m_swap_sizes = true;
                 break;
             default:
                 break;
         }
 
+        if (m_swap_sizes != before_update_swap_sizes) {
+            delete_postprocessing_texture();
+        }
+
         prepare_post_processing_rendering();
```

Comment:
I would pass orientation to prepare and the switch above extracted to the function since the usage of class member variable looks excessive.